### DISCUSSION TITLE
Change default config?

### DISCUSSION THREAD

**Discussion by cyrneko** [YaLTeR/niri#625] [Q&A] [answered]
Hi there! I'm currently building my own personal universal-blue image, with the Niri repos added, and I wanted to add my configuration directly as the default configuration into the system image.

The documentation suggests that Niri never reads a default configuration from a place such as i.e `/etc/xdg/niri`, and will instead dump the default configuration into `$XDG_USER_HOME/niri` / `$HOME/.config/niri`.

On top of that I've found no place I could override the resources (e.g change [this](https://github.com/YaLTeR/niri/blob/main/resources/default-config.kdl))

Can anyone give me a pointer?

**Comment by YaLTeR** [comment]
Hey! Currently, the default config from resources is compiled into the niri binary, and it will only read the config from `~/.config/niri`. I understand it may need some changes for a "fedora spin" like distro packaging, but I don't know what requirements exactly there are.

**Reply by cyrneko** [reply]
Hm, makes sense

I believe just having a system-wide place, e.g `/etc/xdg/niri` or `/etc/niri` to place configs in would suffice, at least for the usecase of shipping another default configuration

I guess the order would then be `/etc[/xdg]/niri` > `~/.config/niri`?

This is how i.e Fedora ships a lightly customized Sway config in `/etc/sway/config` and `/etc/sway/config.d` that is loaded by default unless the user defines their own config in `~/.config/sway`

**Reply by YaLTeR** [reply, marked as answer]
So:
1. If `~/.config/niri/config.kdl` exists, load that.
2. If `/etc/niri/config.kdl` exists, load that.
3. If neither of these exist, write the built-in default config into `~/.config/niri/config.kdl`.

Something like this?

**Reply by cyrneko** [reply]
Yeah, I think that should work!

Backwards-compatible as it doesn't break the previous auto-generating behavior whilst also allowing to override the default configuration :)

**Reply by YaLTeR** [reply]
I have just implemented and pushed this logic, please tell if it works for you.

**Reply by cyrneko** [reply]
> I have just implemented and pushed this logic, please tell if it works for you.

sure thing! I'll be switching to niri-git soon and then test :)

any breaking changes that'd make rolling back to niri 0.1.8 difficult?

**Reply by YaLTeR** [reply]
I don't think so

**Reply by cyrneko** [reply]
update: the behavior works as intended :)

just rebuilt with `niri-git` and tested it out on my laptop, everything went just fine and the config is loaded from /etc/niri/config.kdl
